src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -467,6 +471,73 @@ ucs_status_t uct_cuda_ipc_iface_init_streams(uct_cuda_ipc_iface_t *iface)
     return UCS_OK;
 }
 
+static ucs_status_t
+uct_cuda_ipc_estimate_perf(uct_iface_h tl_iface, uct_perf_attr_t *perf_attr)
+{
+    uct_ep_operation_t op          = UCT_ATTR_VALUE(PERF, perf_attr, operation,
+                                                    OPERATION, UCT_EP_OP_LAST);
+    ucs_memory_type_t src_mem_type = UCT_ATTR_VALUE(PERF, perf_attr,
+                                                    local_memory_type,
+                                                    LOCAL_MEMORY_TYPE,
+                                                    UCS_MEMORY_TYPE_UNKNOWN);
+    ucs_memory_type_t dst_mem_type = UCT_ATTR_VALUE(PERF, perf_attr,
+                                                    remote_memory_type,
+                                                    REMOTE_MEMORY_TYPE,
+                                                    UCS_MEMORY_TYPE_UNKNOWN);
+    int zcopy                      = uct_ep_op_is_zcopy(op);
+    const double latency           = 1.8e-6;
+    const double overhead          = 4.0e-6;
+    int num_nvlinks;
+
+    if (perf_attr->field_mask & UCT_PERF_ATTR_FIELD_BANDWIDTH) {
+        if (uct_ep_op_is_fetch(op)) {
+            /* TODO: differentiate read vs write perf */
+        }
+
+        perf_attr->bandwidth.dedicated = 0;
+        perf_attr->bandwidth.shared    = uct_cuda_base_get_pci_bw(NULL);
+
+        if ((src_mem_type == UCS_MEMORY_TYPE_CUDA) &&
+            (dst_mem_type == UCS_MEMORY_TYPE_CUDA)) {
```

Comment:
could it be if/else? no need to get pci bw for cuda/cuda memory
also, can we cache the pci bw on cuda md?